apps/communications/contacts/js/activities_vcard.js

```diff
@@ -0,0 +1,68 @@
+'use strict';
+/* global ActivityHandler, LazyLoader, VCardReader, Contacts */
+
+ActivityHandler.vCardActivityHandler = {
+  handle: function(activity, handler) {
+    const DEPENDENCIES = '/shared/js/contacts/import/utilities/vcard_reader.js';
+    LazyLoader.load(DEPENDENCIES, function() {
+      getvCardReader(activity.source.data.blob).then(readvCard);
+    });
+
+    function getvCardReader(blob) {
+      return new Promise((resolve) => {
+        var fileReader = new FileReader();
+        fileReader.readAsText(blob);
+        fileReader.onloadend = function() {
+          resolve(new VCardReader(fileReader.result));
+        };
+      });
+    }
+
+    function readvCard(vCardReader) {
+      var firstContact;
+      var cursor = vCardReader.getAll();
+      cursor.onsuccess = function(event) {
+        var contact = event.target.result;
+        // We check if there is only one contact to know what function
+        // we should call. If not, we render the contacts one by one.
+        if (contact) {
+          if (!firstContact) {
+            firstContact = contact;
+            cursor.continue();
+          } else {
+            renderContacts([firstContact, contact], activity, cursor);
+          }
+        } else if (firstContact) {
+          renderOneContact(firstContact, activity);
+        }
+      };
+    }
+
+    function renderOneContact(contact, activity) {
+      handler.mozContactParam = contact;
+      var data = activity.source.data;
+      data.params = {'mozContactParam': true};
+      if (data.allowSave === true) {
+        handler.launch_activity(activity, 'view-contact-form');
+      } else {
+        handler.launch_activity(activity, 'view-contact-details');
+      }
+    }
+
+    function renderContacts(contacts, activity, cursor) {
+      handler.launch_activity(activity, 'multiple-select-view');
+      const DEPENDENCIES = [
+        '/shared/js/contacts/import/utilities/status.js',
+        '/shared/js/simple_phone_matcher.js',
+        '/shared/js/contacts/contacts_matcher.js',
+        '/shared/js/contacts/contacts_merger.js',
+        '/shared/js/contacts/merger_adapter.js',
+        '/contacts/js/views/multiple_select.js'
```

Comment:
this particular dependency should be loaded by the activities.js module as it will be in charge of showing the multiple select view